src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/RazorLoggerFactory.cs

```diff
@@ -2,18 +2,19 @@
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
-using System.ComponentModel.Composition;
+using System.Composition;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.CodeAnalysis.Razor.Logging;
 
-[Export(typeof(IRazorLoggerFactory)), System.Composition.Shared]
+[Shared]
+[Export(typeof(IRazorLoggerFactory))]
 internal class RazorLoggerFactory : IRazorLoggerFactory
 {
     private readonly ILoggerFactory _loggerFactory;
 
     [ImportingConstructor]
-    public RazorLoggerFactory([ImportMany(typeof(IRazorLoggerProvider))] IEnumerable<IRazorLoggerProvider> razorLoggerProviders)
+    public RazorLoggerFactory([ImportMany] IEnumerable<IRazorLoggerProvider> razorLoggerProviders)
```

Comment:
Another MEF mind blowing - had no idea the type would be derived correctly for these.

Reply:
The way it was written before was stating the MEF contract twice. You only need to state the contract in the attribute if it differs from the parameter type. For example, in VS, you might write `[Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider` because the type used for the service contract differs from the actual parameter type.